### PR TITLE
Rados TFA fixes - Online reads balancer fixes with require_min_compat_client param

### PR DESCRIPTION
**Pull request by pdhiran**
Fixed the automation TFA issues. 

There were new features introduced and behaviour was modified post bugs. With the new changes, require_min_compat_client was enforced and no Pg_upmap_primary or MSR rules could be created if the  min_compat_client was not reef & squid resp.

1. Updated the workflow of Online commands and offline commands test scenario for reads balancer.
2. Updated the workflow for Balancer module based read balancing.
3. Updated workflow for MSR EC pool tests.

### CODE REVIEW

File: tests/rados/test_ec_86.py [hunk 1]
```diff
@@ -30,11 +30,86 @@ def run(ceph_cluster, **kw):
...
+                log.debug(
+                    f"require_min_compat_client changed during -ve test the tests is {new_version}"
+                )
+                failed = True
+                pool_exists = True if pool_name in rados_obj.list_pools() else False
+
+                # This check is added now as if the cluster already has MSR pools
+                # min compact client cannot be modified. and we would hit the exception in the code earlier.
+                # if the mode was already upmap-read, the min_compact client could not have been changed.
+                if pool_exists and new_version == "quincy":
```
Comment (pdhiran) [inline tests/rados/test_ec_86.py:83]: version should be reef

File: tests/rados/test_online_reads_balancer.py [hunk 2]
```diff
@@ -75,51 +77,91 @@ def run(ceph_cluster, **kw):
...
+                    failed = True
+                else:
+                    log.info(
+                        f"Expected exception caught: {error_message}. Verifying current balancer mode..."
+                    )
+                    # Fetch current balancer status
+                    balancer_status = rados_obj.run_ceph_command(
+                        cmd="ceph balancer status"
+                    )
+                    if balancer_status.get("mode") == "upmap-read":
```
Comment (pdhiran) [inline tests/rados/test_online_reads_balancer.py:125]: update the param with `balancer_mode`

Comment (pdhiran) [reply tests/rados/test_online_reads_balancer.py:125]: Also add failed = True here.

**Comment by pdhiran** [comment]
pass log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-H2WJS5/

**Comment by pdhiran** [comment]
Pass log : http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-GZV79A

File: tests/rados/test_ec_86.py (hunk 1 above)
Comment (pdhiran) [reply tests/rados/test_ec_86.py:83]: updated

File: tests/rados/test_online_reads_balancer.py (hunk 2 above)
Comment (pdhiran) [reply tests/rados/test_online_reads_balancer.py:125]: updated

**Review by harshkumarRH — APPROVED** [review]
Looks good

**Review by neha-gangadhar — APPROVED** [review]
Approved

**Pull request merged**